synapse_rock/rockcraft.yaml

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: synapse
+summary: Synapse rock
+description: Synapse OCI image for the Synapse charm
+version: "1.0"
+base: bare
+build-base: ubuntu:22.04
+license: Apache-2.0
+platforms:
+  amd64:
+parts:
+    synapse:
+        build-packages:
+            - build-essential
+            - curl
+            - git
+            - libffi-dev
+            - libssl-dev
+            - pkg-config
+            - libjpeg-dev
+            - libpq-dev
+            - libwebp-dev
+            - libxml++2.6-dev
+            - libxslt1-dev
+            - openssl
+            - zlib1g-dev
+            - libicu-dev
+            - python3-dev
+            - python3-pip
+            - python3-setuptools
+        stage-packages:
+            - bash
+            - build-essential
```

Comment:
Do you need to stage `build-essential` ?

Reply:
Same comment for the `-dev` deps, they're usually only needed when you build

Reply:
The final stage in the dockerfile doesn't install all these:
https://github.com/matrix-org/synapse/blob/develop/docker/Dockerfile#L168-L181

Reply:
Removed, thanks